providers/oidc.go

```diff
@@ -238,37 +240,41 @@ func (p *OIDCProvider) findVerifiedIDToken(ctx context.Context, token *oauth2.To
 // buildSessionFromClaims uses IDToken claims to populate a fresh SessionState
 // with non-Token related fields.
 func (p *OIDCProvider) buildSessionFromClaims(idToken *oidc.IDToken) (*sessions.SessionState, error) {
-	newSession := &sessions.SessionState{}
+	ss := &sessions.SessionState{}
 
 	if idToken == nil {
-		return newSession, nil
+		return ss, nil
 	}
 
 	claims, err := p.getClaims(idToken)
 	if err != nil {
 		return nil, fmt.Errorf("couldn't extract claims from id_token (%v)", err)
 	}
 
-	newSession.User = claims.Subject
-	newSession.Email = claims.Email
-	newSession.Groups = claims.Groups
-	newSession.PreferredUsername = claims.PreferredUsername
+	ss.User = claims.Subject
+	ss.Email = claims.Email
+	ss.Groups = claims.Groups
+
+	// TODO (@NickMeves) Deprecate for dynamic claim to session mapping
+	if pref, ok := claims.rawClaims["preferred_username"].(string); ok {
+		ss.PreferredUsername = pref
+	}
```

Comment:
This reminds me of one thought I had before, I wonder if we could just use the easyjson library to extract the claims 🤔 We could come up with some helper for all available claims and set them appropriately using some mapping